pkg/bpf/prog.go

```diff
@@ -67,3 +74,92 @@ func (t ProgType) String() string {
 
 	return "Unknown"
 }
+
+// attrProg holds values from the upstream struct union for BPF_*_GET_*_ID.
+// From: https://github.com/torvalds/linux/blob/v4.16/include/uapi/linux/bpf.h#L299
+type attrProg struct {
+	progID    uint32 // union: start_id, prog_id, or map_id
+	nextID    uint32
+	openFlags uint32
+}
+
+// ProgInfo holds values from the upstream struct bpf_prog_info.
+// From: https://github.com/torvalds/linux/blob/v4.16/include/uapi/linux/bpf.h#924
+type ProgInfo struct {
+	ProgType        uint32
+	ID              uint32
+	Tag             uint8
+	JitedProgLen    uint32
+	XlatedProgLen   uint32
+	JitedProgInsns  uint64
+	XlatedProgInsns uint64
+	LoadTime        uint64 // ns since boottime
+	CreatedByUID    uint32
+	NRMapIDs        uint32
+	MapIDs          []uint32
+	Name            string
+	IfIndex         uint32
+	NetnsDev        uint64
+	NetnsIno        uint64
+}
+
+// attrObjInfo holds values from the upstream struct union for BPF_OBJ_GET_INFO_BY_FD.
+// From: https://github.com/torvalds/linux/blob/v4.16/include/uapi/linux/bpf.h#L309
+type attrObjInfo struct {
+	bpfFD   uint32
+	infoLen uint32
+	info    uint64
+}
+
+// GetProgNextID takes a current program ID and a pointer to the next program ID
+// and fills the next program ID.
+func GetProgNextID(current uint32, next *uint32) error {
```

Comment:
It occurred to me that my C background was showing through on my prior review here. In Go it's more natural to return `(uint32, error)` rather than passing in a pointer.